crates/polars-core/src/chunked_array/ops/arity.rs

```diff
@@ -27,6 +37,72 @@ impl<A1, A2, R, T: FnMut(A1, A2) -> R> BinaryFnMut<A1, A2> for T {
     type Ret = R;
 }
 
+#[inline]
+pub fn unary_elementwise<'a, T, V, F>(ca: &'a ChunkedArray<T>, mut op: F) -> ChunkedArray<V>
+where
+    T: PolarsDataType,
+    V: PolarsDataType,
+    F: UnaryFnMut<Option<T::Physical<'a>>>,
+    V::Array: ArrayFromIter<<F as UnaryFnMut<Option<T::Physical<'a>>>>::Ret>,
+{
```

Comment:
Can you add a check if `ca.len() == ca.null_count()` and if so make an array filled with `op(None)` values instead of evaluating `op` a lot?